wayland: match x11 cursor behavior

Dudemanguy911 opened the pull request:
This is forked from #6480 with just a few more lines to fix what I had mentioned there. I only made this since there didn't seem to be any activity from @t-8ch. This also fixes #6185. On wayland, the cursor should automatically hide without having to move it (which is the current behavior). Also, the cursor shouldn't reappear when going to fullscreen (which is what happened with @t-8ch's fix).

Also as a side note, I personally don't really think it's necessary to define a whole new `reset_cursor` function. I think the name is misleading and `set_cursor_visibility(wl, wl->cursor_visible)` is more than clear and sufficient. I left `reset_cursor` since like 50% of this is stolen from @t-8ch anyway, but I'll happily change it if the devs agree with me.

I agree that my changes can be relicensed to LGPL 2.1 or later.

haasn commented on video/out/wayland_common.c:
```diff
@@ -91,6 +92,11 @@ static int set_cursor_visibility(struct vo_wayland_state *wl, bool on)
 
+static void reset_cursor(struct vo_wayland_state *wl)
+{
+        set_cursor_visibility(wl, wl->cursor_visible);
```
Why add a 1-line function that you're only going to call once?

Dudemanguy911 replied:
I totally agree. Like I said earlier, I only left it in since it was originally forked from t-8ch's PR. I'll happily remove it.

zc62 commented:
Maybe better to edit the original commit so that the one line function never appears in the git history, rather than adding an extra commit to manually delete it later.

Dudemanguy911 commented:
The idea wasn't originally mine though (the reason why I forked it from the other PR to begin with), so I don't think it's appropriate to rewrite history like that.

zc62 commented:
It is not uncommon to slightly modify the commit while maintaining the original authorship of the that commit. I don't see why it is inappropriate to do that. Anyway, this is just my point of view. It's up to the members to decide if this is okay.

Dudemanguy911 commented:
Fair enough point. Personally, I prefer not to modify other people's commits directly, but certainly there are people that do modify them directly. And yeah, I'll happily go with whatever approach is preferred by mpv devs.

Dudemanguy911 commented:
Revisiting this, I changed my mind and figured zc62 was right. It looks much cleaner. I went ahead and squashed those commits so now it's just two. Pinging @haasn and/or @atomnuker to take another look at this.

t-8ch commented:
@jeeb Any chance you could have a look at this?
